otter/convergence/service.py

```diff
@@ -0,0 +1,54 @@
+"""Converger service"""
+
+from effect.twisted import perform
+
+from twisted.application.service import Service
+
+from otter.constants import LOCK_PATH
+from otter.convergence.composition import (
+    execute_convergence, get_desired_group_state)
+from otter.util.deferredutils import with_lock
+
+
+class Converger(Service, object):
+    """Converger service"""
+
+    def __init__(self, reactor, kz_client, dispatcher):
+        self._reactor = reactor
+        self._kz_client = kz_client
+        self._dispatcher = dispatcher
+
+    def _get_lock(self, group_id):
+        """Get a ZooKeeper-backed lock for converging the group."""
+        path = '{}/convergence/{}'.format(LOCK_PATH, group_id)
+        return self._kz_client.Lock(path)
+
+    def converge(self, group_id, desired, launch_config,
+                 perform=perform,
+                 execute_convergence=execute_convergence):
+        """Converge a group to a capacity with a launch config."""
+        def exec_convergence():
+            desired_group_state = get_desired_group_state(
```

Comment:
`Converger` should get desired capacity and launch config from CASS at the beginning of the convergence cycle as it can change in the process. @lvh was working on that in #956 (I think). Not sure if you want to do it in this PR.